src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -0,0 +1,644 @@
+import org.prebid.server.auction.model.BidRejectionTracker;
+import org.prebid.server.json.EncodeException;
+import org.prebid.server.model.HttpRequestContext;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.version.PrebidVersionProvider;
+import org.prebid.server.vertx.httpclient.HttpClient;
+import org.prebid.server.vertx.httpclient.model.HttpClientResponse;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.Clock;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class GreenbidsAnalyticsReporterTest extends VertxTest {
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private HttpClient httpClient;
+
+    @Mock
+    private Clock clock;
+
+    @Mock
+    private PrebidVersionProvider prebidVersionProvider;
+
+    @Captor
+    private ArgumentCaptor<String> jsonCaptor;
+
+    @Captor
+    private ArgumentCaptor<MultiMap> headersCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> urlCaptor;
+
+    @Captor
+    private ArgumentCaptor<Long> timeoutCaptor;
+
+    private GreenbidsAnalyticsReporter target;
+
+    private GreenbidsAnalyticsProperties greenbidsAnalyticsProperties;
+
+    private GreenbidsJacksonMapper jacksonMapper;
+
+    @Before
+    public void setUp() {
+        final ObjectMapper mapper = new ObjectMapper();
+        jacksonMapper = new GreenbidsJacksonMapper(mapper);
+
+        greenbidsAnalyticsProperties = GreenbidsAnalyticsProperties.builder()
+                .exploratorySamplingSplit(0.9)
+                .analyticsServerVersion("2.2.0")
+                .analyticsServerUrl("http://localhost:8080")
+                .configurationRefreshDelayMs(10000L)
+                .timeoutMs(100000L)
+                .build();
+
+        target = new GreenbidsAnalyticsReporter(
+                greenbidsAnalyticsProperties,
+                jacksonMapper,
+                httpClient,
+                clock,
+                prebidVersionProvider);
+
+        jsonCaptor = ArgumentCaptor.forClass(String.class);
+        headersCaptor = ArgumentCaptor.forClass(MultiMap.class);
+        urlCaptor = ArgumentCaptor.forClass(String.class);
+        timeoutCaptor = ArgumentCaptor.forClass(Long.class);
```

Comment:
as I remember the Captor annotation should init the captors automatically, maybe some tweaks to the Rule is required for that